libraries/AP_GPS/AP_GPS_NOVA.cpp

```diff
@@ -276,10 +315,55 @@ AP_GPS_NOVA::process_message(void)
         const psrdop &psrdopu = nova_msg.data.psrdopu;
 
         state.hdop = (uint16_t) (psrdopu.hdop*100);
-        state.vdop = (uint16_t) (psrdopu.htdop*100);
+        //state.vdop = (uint16_t) (psrdopu.htdop*100);
+        state.vdop = (uint16_t)((sqrt(psrdopu.pdop - psrdopu.hdop)) * 100);
```

Comment:
this risks producing the sqrt of a negative number. I'd suggest leaving it as the htdop as the vdop is not a critical number in any calculations (its not used by the EKF)